### PR TITLE
Add sample for validator

### PR DESCRIPTION
**Pull request by mloufra**
Signed-off-by: Frank Lou <mloufra@amazon.com>

### Description
Create a sample Setting in Hello World Extension package, and using the RegexValidator form the Setting on Opensearch side the validate the value of sample Setting.

Needs [OS #6221](https://github.com/opensearch-project/OpenSearch/pull/6221) to be merged first.

### Issues Resolved
Fixes https://github.com/opensearch-project/opensearch-sdk-java/issues/350

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/sdk/sample/helloworld/HelloWorldExtension.java
Comment (dbwiddis) [inline src/main/java/org/opensearch/sdk/sample/helloworld/HelloWorldExtension.java:40]: Please put all the new setting code in a new class, not the top level extension.   The `ExampleCustomSettingsConfig` in the plugin we're modeling is a good class name unless you have a better idea.

In this class, you will add the `getSettings()` extension point referencing this setting in the new class.

**Review by dbwiddis — CHANGES_REQUESTED** [review]
Good start, see comments inline.

File: src/main/java/org/opensearch/sdk/sample/helloworld/HelloWorldExtension.java
Comment (mloufra) [reply src/main/java/org/opensearch/sdk/sample/helloworld/HelloWorldExtension.java:58]: ACK

Comment (mloufra) [reply src/main/java/org/opensearch/sdk/sample/helloworld/HelloWorldExtension.java:58]: ACK

File: src/main/java/org/opensearch/sdk/sample/helloworld/HelloWorldExtension.java
Comment (mloufra) [reply src/main/java/org/opensearch/sdk/sample/helloworld/HelloWorldExtension.java:64]: I will create another class `ExampleCustomSettingsConfig`, and make this method in that class.

File: src/main/java/org/opensearch/sdk/sample/helloworld/HelloWorldExtension.java
Comment (mloufra) [reply src/main/java/org/opensearch/sdk/sample/helloworld/HelloWorldExtension.java:40]: ACK

File: src/main/java/org/opensearch/sdk/sample/helloworld/HelloWorldExtension.java
Comment (dbwiddis) [reply src/main/java/org/opensearch/sdk/sample/helloworld/HelloWorldExtension.java:64]: You don't need this method at all. You just need to instantiate your validator.

Comment (mloufra) [reply src/main/java/org/opensearch/sdk/sample/helloworld/HelloWorldExtension.java:64]: got it

File: src/main/java/org/opensearch/sdk/sample/helloworld/ExampleCustomSettingConfig.java
Comment (dbwiddis) [inline src/main/java/org/opensearch/sdk/sample/helloworld/ExampleCustomSettingConfig.java:21]: Code looks great except for this variable which is never set. You have a getter for it, though! 

Both can probably be removed.

Comment (mloufra) [reply src/main/java/org/opensearch/sdk/sample/helloworld/ExampleCustomSettingConfig.java:21]: ACK

File: src/main/java/org/opensearch/sdk/sample/helloworld/ExampleCustomSettingConfig.java
Comment (dbwiddis) [inline src/main/java/org/opensearch/sdk/sample/helloworld/ExampleCustomSettingConfig.java:37]: Rather than a getter, I'd suggest just making the constant itself `public` at line 25 so you don't need this.

**Review by dbwiddis — APPROVED** [review]
LGTM (pending merge on opensearch side) but one improvement suggested.

File: src/main/java/org/opensearch/sdk/sample/helloworld/ExampleCustomSettingConfig.java
Comment (mloufra) [reply src/main/java/org/opensearch/sdk/sample/helloworld/ExampleCustomSettingConfig.java:37]: ACK

File: src/main/java/org/opensearch/sdk/sample/helloworld/ExampleCustomSettingConfig.java
Comment (owaiskazi19) [inline src/main/java/org/opensearch/sdk/sample/helloworld/ExampleCustomSettingConfig.java:23]:
```suggestion
     * A string setting. If the string setting matches the FORBIDDEN_REGEX string, the validation will fail.
```

File: src/main/java/org/opensearch/sdk/sample/helloworld/ExampleCustomSettingConfig.java
Comment (owaiskazi19) [inline src/main/java/org/opensearch/sdk/sample/helloworld/ExampleCustomSettingConfig.java:26]:
```suggestion
        "custom.validate",
```

File: src/main/java/org/opensearch/sdk/sample/helloworld/HelloWorldExtension.java
Comment (owaiskazi19) [inline src/main/java/org/opensearch/sdk/sample/helloworld/HelloWorldExtension.java:56]: Need Javadoc

**Review by owaiskazi19 — APPROVED** [review]
Small nits. Rest LGTM!

File: src/main/java/org/opensearch/sdk/sample/helloworld/HelloWorldExtension.java
Comment (mloufra) [reply src/main/java/org/opensearch/sdk/sample/helloworld/HelloWorldExtension.java:59]: ACK

File: src/main/java/org/opensearch/sdk/sample/helloworld/ExampleCustomSettingConfig.java
Comment (mloufra) [reply src/main/java/org/opensearch/sdk/sample/helloworld/ExampleCustomSettingConfig.java]: ACK

File: src/main/java/org/opensearch/sdk/sample/helloworld/ExampleCustomSettingConfig.java
Comment (mloufra) [reply src/main/java/org/opensearch/sdk/sample/helloworld/ExampleCustomSettingConfig.java]: ACK

**Review by dbwiddis — APPROVED** [review]
LGTM (don't merge until [OS #6221](https://github.com/opensearch-project/OpenSearch/pull/6221) is merged)

**Comment by ryanbogan** [comment]
@mloufra Can you rebase this PR with main?

**Comment by mloufra** [comment]
> 

sure

**Comment by mloufra** [comment]
> @mloufra Can you rebase this PR with main?

Done

**Review by ryanbogan — APPROVED** [review]

**Review by joshpalis — APPROVED** [review]

**Pull request merged**
